app/src/sharedTest/java/org/oppia/app/RecyclerViewMatcher.kt

```diff
@@ -0,0 +1,54 @@
+package org.oppia.app
+
+import android.content.res.Resources
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+
+class RecyclerViewMatcher(private val recyclerViewId: Int) {
```

Comment:
Could we maybe use the one in BindableAdapterTest, instead? Prefer extracting that one to here for reuse since I think that one has all the necessary functionality, and is simpler.

Also, please add documentation and credit links where applicable (if you didn't derive this solution on your own).

Reply:
I was not able to use code from `BindableAdapterTest `, but I have updated the code for better readability.

Reply:
Also, I have created a custom adapter for this TopicTrainFragment, so will the `BindableAdapterTest ` work with this?